flow360/component/flow360_params/flow360_temp.py

```diff
@@ -0,0 +1,123 @@
+
+    LEFT_HAND = "leftHand"
+    RIGHT_HAND = "rightHand"
+
+
+class BETDiskTwist(Flow360BaseModel):
+    """:class:`BETDiskTwist` class"""
+
+    radius: Optional[float] = pd.Field()
+    twist: Optional[float] = pd.Field()
+
+
+class BETDiskChord(Flow360BaseModel):
+    """:class:`BETDiskChord` class"""
+
+    radius: Optional[float] = pd.Field()
+    chord: Optional[float] = pd.Field()
+
+
+class BETDiskSectionalPolar(Flow360BaseModel):
+    """:class:`BETDiskSectionalPolar` class"""
+
+    lift_coeffs: Optional[List[List[List[float]]]] = pd.Field(alias="liftCoeffs")
+    drag_coeffs: Optional[List[List[List[float]]]] = pd.Field(alias="dragCoeffs")
+
+
+class BETDisk(Flow360BaseModel):
+    """:class:`BETDisk` class"""
+
+    rotation_direction_rule: Optional[RotationDirectionRule] = pd.Field(
+        alias="rotationDirectionRule"
+    )
+    center_of_rotation: Coordinate = pd.Field(alias="centerOfRotation")
+    axis_of_rotation: Axis = pd.Field(alias="axisOfRotation")
+    number_of_blades: PositiveInt = pd.Field(alias="numberOfBlades")
+    radius: PositiveFloat = pd.Field()
+    omega: float = pd.Field()
+    chord_ref: PositiveFloat = pd.Field(alias="chordRef")
+    thickness: PositiveFloat = pd.Field(alias="thickness")
+    n_loading_nodes: PositiveInt = pd.Field(alias="nLoadingNodes")
+    blade_line_chord: Optional[PositiveFloat] = pd.Field(alias="bladeLineChord")
+    initial_blade_direction: Optional[Coordinate] = pd.Field(alias="initialBladeDirection")
+    tip_gap: Optional[Union[NonNegativeFloat, Literal["inf"]]] = pd.Field(alias="tipGap")
+    mach_numbers: List[float] = pd.Field(alias="MachNumbers")
+    reynolds_numbers: List[PositiveFloat] = pd.Field(alias="ReynoldsNumbers")
+    alphas: List[float] = pd.Field()
+    volume_name: Optional[str] = pd.Field(alias="volumeName")
+    twists: List[BETDiskTwist] = pd.Field()
+    chords: List[BETDiskChord] = pd.Field()
+    sectional_polars: List[BETDiskSectionalPolar] = pd.Field(alias="sectionalPolars")
+    sectional_radiuses: List[float] = pd.Field(alias="sectionalRadiuses")
+
+
+class PorousMediumVolumeZone(Flow360BaseModel):
+    """:class:`PorousMediumVolumeZone` class"""
+
+    zone_type: str = pd.Field(alias="zoneType")
+    center: Coordinate = pd.Field()
+    lengths: Coordinate = pd.Field()
+    axes: List[Coordinate] = pd.Field(min_items=2, max_items=3)
+    windowing_lengths: Optional[Coordinate] = pd.Field(alias="windowingLengths")
+
+
+class PorousMedium(Flow360BaseModel):
+    """:class:`PorousMedium` class"""
+
+    darcy_coefficient: Coordinate = pd.Field(alias="DarcyCoefficient")
+    forchheimer_coefficient: Coordinate = pd.Field(alias="ForchheimerCoefficient")
+    volume_zone: PorousMediumVolumeZone = pd.Field(alias="volumeZone")
+
+
+class UserDefinedDynamic(Flow360BaseModel):
+    """:class:`UserDefinedDynamic` class"""
+
+    dynamics_name: str = pd.Field(alias="dynamicsName")
+    input_vars: List[str] = pd.Field(alias="inputVars")
```

Comment:
see here: https://docs.flexcompute.com/projects/flow360/en/latest/solverConfiguration/solverConfiguration.html#userdefineddynamics

List of the inputs to define the user defined dynamics. Supported inputs are: "CL", "CD", "torque_bet_NUM", "thrust_bet_NUM", (NUM is the index of the BET disk starting from 0), "rotMomentX", "rotMomentY", "rotMomentZ" (X/Y/Z moments with respect to [momentCenter](https://docs.flexcompute.com/projects/flow360/en/latest/solverConfiguration/solverConfiguration.html#geometryconfiguration)), "forceX", "forceY", "forceZ"

Not sure how to dynamically implement thrust_bet_NUM. We might need custom validation for it

Reply:
1. use Literals for defined variables
2. use 'str' for torque_bet_NUM, with 2 validators:
 - regex on assignment
 - range validator on to_solver() method - to do when introduced dimensioned types

Reply:
Ignored for now - UserDefinedDynamics is a beta feature